telethon/tl/custom/chatgetter.py

```diff
@@ -10,15 +11,20 @@ class ChatGetter(abc.ABC):
     and `chat_id` properties and `get_chat` and `get_input_chat`
     methods.
     """
-    def __init__(self, chat_peer=None, *, input_chat=None, chat=None, broadcast=None):
+    def __init__(self,
+                 chat_peer: Optional[types.TypePeer] = None,
+                 *,
+                 input_chat: Optional[types.Chat] = None,
+                 chat: Optional[types.Chat] = None,
+                 broadcast: Optional[bool] = None):
         self._chat_peer = chat_peer
         self._input_chat = input_chat
         self._chat = chat
         self._broadcast = broadcast
         self._client = None
 
     @property
-    def chat(self):
+    def chat(self) -> types.Chat:
```

Comment:
See below:

> ```
> Returns the :tl:`User`, :tl:`Chat` or :tl:`Channel` where this object
> ```

This should probably be an union. And maybe Optional too, since it may be missing.

Reply:
Maybe it needs a new type in the `types` module? I couldn't find any premade type containing all of those 3 types. and using unions all over the place is a little bit dirty, isn't it? (The TypePeer is actually well-fitting, except that it is for peers: `TypePeer = Union[PeerUser,PeerChat,PeerChannel]`)

Reply:
Yeah, a union could be added to `hints.py` for this.